gke-deploy/core/image/image.go

```diff
@@ -83,17 +77,14 @@ func GetName(ctx context.Context, image Image) (string, error) {
 }
 
 // GetDigest gets an image's corresponding digest.
-func GetDigest(ctx context.Context, image Image, gs services.GcloudService) (string, error) {
-	switch t := image.(type) {
-	case name.Tag:
-		digest, err := gs.ContainerImagesDescribe(ctx, t.Name(), imageDigestFormat)
-		if err != nil {
-			return "", fmt.Errorf("failed to get image digest: %v", err)
-		}
-		return digest, nil
-	case name.Digest:
-		return image.Identifier(), nil
-	default:
-		return "", fmt.Errorf("invalid image type: %s", t)
+func GetDigest(ctx context.Context, image name.Reference, rs services.RemoteService) (string, error) {
```

Comment:
I'd suggest something like `ResolveReference`, since this returns a fully-qualified string containing the image name and digest, not just the digest.

Reply:
This only returns the digest

Reply:
But, looking at my code, it can return the fully-qualified string rather than just the digest. I will change this.